sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeContractQuery.java

```diff
@@ -0,0 +1,314 @@
+/*-
+ *
+ * Hedera Java SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.hedera.hashgraph.sdk;
+
+import static com.hedera.hashgraph.sdk.EntityIdHelper.getContractAddressFromMirrorNodeAsync;
+import static com.hedera.hashgraph.sdk.EntityIdHelper.getEvmAddressFromMirrorNodeAsync;
+import static com.hedera.hashgraph.sdk.EntityIdHelper.performQueryToMirrorNodeAsync;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.protobuf.ByteString;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import org.bouncycastle.util.encoders.Hex;
+
+/**
+ * MirrorNodeContractQuery returns a result from EVM execution such as cost-free execution of read-only smart contract
+ * queries, gas estimation, and transient simulation of read-write operations.
+ */
+public class MirrorNodeContractQuery {
+    // The contract we are sending the transaction to
+    private ContractId contractId = null;
+    private String contractEvmAddress = null;
+    // The account we are sending the transaction from
+    private AccountId sender = null;
+    private String senderEvmAddress = null;
+    // The transaction callData
+    private byte[] callData;
+    // The amount we are sending to payable functions
```

Comment:
We can send value on a contract deployment call not only payable functions, so the code comment can be adjusted here